flink-runtime/src/test/java/org/apache/flink/runtime/taskexecutor/slot/TaskSlotTableTest.java

```diff
@@ -243,8 +252,181 @@ public void testGenerateSlotReport() throws SlotNotFoundException {
 		}
 	}
 
+	@Test
+	public void testAllocateSlot() {
+		final JobID jobId = new JobID();
+		final AllocationID allocationId = new AllocationID();
+		TaskSlotTable taskSlotTable = null;
+		try {
+			taskSlotTable = createTaskSlotTableWithAllocatedSlot(
+				jobId,
+				allocationId,
+				new TestingSlotActionsBuilder().build());
+			Iterator<TaskSlot> allocatedSlots = taskSlotTable.getAllocatedSlots(jobId);
+			TaskSlot nextSlot = allocatedSlots.next();
+			assertThat(nextSlot.getIndex(), is(0));
+			assertThat(nextSlot.getAllocationId(), is(allocationId));
+			assertThat(nextSlot.getJobId(), is(jobId));
+			assertThat(allocatedSlots.hasNext(), is(false));
+		} finally {
+			stopTable(taskSlotTable);
+		}
+	}
+
+	@Test
+	public void testAddTask() throws Exception {
+		final JobID jobId = new JobID();
+		final ExecutionAttemptID executionAttemptId = new ExecutionAttemptID();
+		final AllocationID allocationId = new AllocationID();
+		Task task = createTask(
+			jobId,
+			executionAttemptId,
+			allocationId,
+			EmptyInvokable.class);
+		TaskSlotTable taskSlotTable = null;
+		try {
+			taskSlotTable = createTaskSlotTableWithStartedTask(task, new TestingSlotActionsBuilder().build());
+			Iterator<Task> tasks = taskSlotTable.getTasks(jobId);
+			Task nextTask = tasks.next();
+			assertThat(nextTask.getExecutionId(), is(executionAttemptId));
+			assertThat(nextTask.getAllocationId(), is(allocationId));
+			assertThat(tasks.hasNext(), is(false));
+			taskSlotTable.freeSlot(allocationId);
+			taskSlotTable.removeTask(executionAttemptId);
```

Comment:
why are these calls necessary?

Reply:
I removed the unnecessary calls, also for subsequent comments and added comment in code for one where I think it was needed.